### DISCUSSION TITLE
Can I pass parameter from iframe to embed app?

### DISCUSSION THREAD

**Discussion by hunter-xue** [Budibase/budibase#13381] [Help] [answered]
I created an embed app, and this app need several paramters from my "main" app, I tried use src of iframe like this: 
`src="http://aabbcc.com:8080/embed/my-first-app?foo=bar"`
but "foo=bar" cannot be read in screan load. 
Any opinion? thanks.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @hunter-xue,

I had an extensive look into this and it will require snippets to achieve. JS within the design screen is sandboxed meaning you won't be able to get the `window.location.href`. What this does is gets the current page url. Without it you won't be able to get parameters attached to the URL.

The below code should work if used within a [snippet](https://docs.budibase.com/docs/snippets) and then referenced inside a bindable area.

```js
return function (urlString) {
    const url = window.location.href;
    const urlObject = new URL(url);
    const exampleParamValue = urlObject.searchParams.get('example');

    return exampleParamValue;
}
```

Paragraph component that references the snippet is shown in the below image.
<img width="1511" alt="Screenshot 2024-04-04 at 13 36 46" src="https://github.com/Budibase/budibase/assets/126772285/77d1008b-3f35-4b2c-a68a-dd34b85cef71">

**Demo of this working**
![Untitled_Project_V1](https://github.com/Budibase/budibase/assets/126772285/54a6f6c5-da1b-4c33-8bd9-6d61f39d7ea7)

You can find more information on snippets [here](https://docs.budibase.com/docs/snippets).

**Reply by hunter-xue** [reply]
pretty cool~~~, that's really a good idea👍
